### PR TITLE
cut 1.7.0

### PR DESCRIPTION
**Pull request by brancz**
We had a lot of patch releases in the 1.6.x release series, so I'm not sure whether I caught all relevant changes in comparison to 1.6.3.

Please comment if I forgot anything or should describe differently.

@beorn7 @brian-brazil @fabxc @grobie @juliusv @tomwilkie

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+## 1.7.0 / 2017-06-06
+
+* [CHANGE] Compress remote storage requests and responses.
```
Comment (brian-brazil) [inline CHANGELOG.md]: They were already compressed, we changed the framing

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+* [FEATURE] Add OpenStack service discovery.
+* [FEATURE] Add ability to limit Kubernetes service discovery to certain namespaces.
+* [FEATURE] Add metric for discovered number of Alertmanagers.
+* [ENHANCEMENT] Print system information (uname) on start up.
+* [ENHANCEMENT] Properly ellide secrets in config.
```
Comment (brian-brazil) [inline CHANGELOG.md]: I think this may qualify as a CHANGE as the config is visible in the UI.

File: CHANGELOG.md (hunk 1 above)
Comment (brancz) [inline CHANGELOG.md]: I see, I didn't follow entirely so I just used the PR title. What would you suggest?

**Comment by brancz** [comment]
@brian-brazil do you think #2766 would be described as an enhancement?

File: CHANGELOG.md (hunk 2 above)
Comment (grobie) [inline CHANGELOG.md]: I don't mind too much, but I'd expect CHANGE only for actually breaking changes. As we don't have an API for the config yet, I wouldn't list pure UI changes.

**Comment by brian-brazil** [comment]
Yeah, I think it's an enhancement.

File: CHANGELOG.md (hunk 2 above)
Comment (brancz) [inline CHANGELOG.md]: I actually changed it now, I don't have a strong opinion on one or the other.

**Comment by grobie** [comment]
Some more things?

* enhancement promtool linter checks counter naming and more reserved labels
* bugfix https://github.com/prometheus/prometheus/pull/2660
* bugfix https://github.com/prometheus/prometheus/pull/2656
* bugfix https://github.com/prometheus/prometheus/pull/2708
* bugfix https://github.com/prometheus/prometheus/pull/2733
* bugfix https://github.com/prometheus/prometheus/pull/2778
* bugfix https://github.com/prometheus/prometheus/pull/2783

**Comment by brancz** [comment]
@grobie added :slightly_smiling_face: .

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,22 @@
+## 1.7.0 / 2017-06-06
+
+* [CHANGE] Compress remote storage requests and responses with unframed/raw snappy.
+* [CHANGE] Properly ellide secrets in config.
+* [FEATURE] Add OpenStack service discovery.
+* [FEATURE] Add ability to limit Kubernetes service discovery to certain namespaces.
+* [FEATURE] Add metric for discovered number of Alertmanagers.
+* [ENHANCEMENT] Print system information (uname) on start up.
+* [ENHANCEMENT] Show gaps in graphs on expression browser.
+* [ENHANCEMENT] Enhancement promtool linter checks counter naming and more reserved labels.
```
Comment (grobie) [inline CHANGELOG.md]: I guess we can remove the `enhancement` here? :)

Comment (brancz) [inline CHANGELOG.md]: oops

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,22 @@
...
+
+* [CHANGE] Compress remote storage requests and responses with unframed/raw snappy.
+* [CHANGE] Properly ellide secrets in config.
+* [FEATURE] Add OpenStack service discovery.
+* [FEATURE] Add ability to limit Kubernetes service discovery to certain namespaces.
+* [FEATURE] Add metric for discovered number of Alertmanagers.
+* [ENHANCEMENT] Print system information (uname) on start up.
+* [ENHANCEMENT] Show gaps in graphs on expression browser.
+* [ENHANCEMENT] Promtool linter checks counter naming and more reserved labels.
+* [BUGFIX] Fix broken mesos discovery.
```
Comment (fabxc) [inline CHANGELOG.md]: Mesos

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,22 @@
...
+* [BUGFIX] Fix redirect when external url is set.
```
Comment (fabxc) [inline CHANGELOG.md]: URL

**Comment by tomwilkie** [comment]
Would it be possible to merge https://github.com/prometheus/prometheus/pull/2786 before me release 1.7.0?  Its been LGTM'd.

**Comment by brancz** [comment]
@tomwilkie merged. Do you think it needs to be mentioned in the release notes?

**Pull request merged**